src/designpatterns/observer/template/observer_template.puml

```diff
@@ -0,0 +1,28 @@
+@startuml
+interface Subject{
+    addObserver(Observer observer): void
+    removeObserver(Observer observer): void
+    notifyObserver(): void
+    setState(Object o): void
+}
+interface Observer{
+    update(Object o): void
+}
+class SubjectImpl implements Subject{
+    - state: Object
+    - observer: Set<Observer>
+    + SubjectImpl()
+    + addObserver(Observer observer): void
+    + removeObserver(Observer observer): void
+    + notifyObserver(): void
+    + setState(Object o): void
+}
+class ObserverImpl implements Observer{
+    - observerName: String
+    + ObserverImpl(observerName: String)
+    + update(Object o)
+}
+'Relationship between classes
+Subject o-- "1..1" Observer : aggregation
```

Comment:
we can use o.. as it is loosely coupled association. Objects are not created within class